build.sh

```diff
@@ -0,0 +1,49 @@
+function isinstalled {
+  if command -v "$1";
+  then
+    true
+  else
+    false
+  fi
+}
+
+# doing this to make this work locally since 99% of the times most dev machines already have curl and might not have apt-get
+if isinstalled "curl"
+then
+  echo "curl already installed moving on"
+else
+  apt-get update && apt install -y curl
+fi
+
+NANCY_VERSION=v0.1.15
+
+mkdir -p /tmp/tools
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  curl -s -L "https://github.com/sonatype-nexus-community/nancy/releases/download/$NANCY_VERSION/nancy-darwin.amd64-$NANCY_VERSION" -o "/tmp/tools/nancy"
+else
+  curl -s -L "https://github.com/sonatype-nexus-community/nancy/releases/download/$NANCY_VERSION/nancy-linux.amd64-$NANCY_VERSION" -o "/tmp/tools/nancy"
+fi
+chmod +x /tmp/tools/nancy
+
+set -e
+DEP_EXIT_CODE=0
+SUM_EXIT_CODE=0
+LIST_EXIT_CODE=0
+
+/tmp/tools/nancy Gopkg.lock || DEP_EXIT_CODE=$?
+/tmp/tools/nancy go.sum || SUM_EXIT_CODE=$?
+GO111MODULE=on go list -m all | /tmp/tools/nancy || LIST_EXIT_CODE=$?
+
+echo "go list exit code : $LIST_EXIT_CODE"
+echo "go sum exit code : $SUM_EXIT_CODE"
+echo "dep exit code : $DEP_EXIT_CODE"
+
+
+if [[ "$LIST_EXIT_CODE" -eq 0 || "$SUM_EXIT_CODE" -eq 0 || "$DEP_EXIT_CODE" -eq 0 ]]
+then
+  echo "One of the above nancy tests actually passed....that means the project is no longer vulenerable and thats an issue"
+  exit 199
```

Comment:
LOL any specific reason for 199?

Reply:
Nope none at all :) I can make it whatever magic number you want :)

Reply:
I'd just use one. Unix catch all for error.

Reply:
Ok here you go 0f1dab2

Reply:
Rad! We were using variable exit codes in code before (returning the number of vulnerable libraries as an exit code), and while it works, if someone were scripting off the exit code, they'd generally expect 1. In this case it doesn't matter, because it's just used to fail, but I try and avoid using anything but `0`, or `1`, unless you want to invent your own code (so someone would know oh OSS Index is down if it exists as `3` or whatever)